configutil/kms.go

```diff
@@ -208,10 +268,35 @@ func configureWrapper(
 		return nil, nil, fmt.Errorf("error parsing config options: %w", err)
 	}
 
+	// If the KMS block contained plugin file information, add it
+	pluginOpts := opts.withPluginOptions
+	switch {
+	case configKMS.PluginPath == "" && configKMS.PluginChecksum == "":
+	case configKMS.PluginPath == "" && configKMS.PluginChecksum != "":
+		return nil, nil, errors.New("plugin checksum specified in kms but plugin path empty")
+	case configKMS.PluginPath != "" && configKMS.PluginChecksum == "" && len(configKMS.pluginChecksumBytes) == 0:
+		return nil, nil, errors.New("plugin path specified in kms but plugin checksum empty")
```

Comment:
What do you think about
```suggestion
	case configKMS.PluginPath == "":
		if configKMS.PluginChecksum != "" {
			return nil, nil, errors.New("plugin checksum specified in kms but plugin path empty")
		}
	case configKMS.PluginChecksum == "" && len(configKMS.pluginChecksumBytes) == 0:
		return nil, nil, errors.New("plugin path specified in kms but plugin checksum empty")
```
instead?

Reply:
Either way. I kind of like when there are two vars to have that explicit possibility matrix enumerated in the cases because I feel like it makes it easier for others to at-a-glace see if all cases have been handled.

Reply:
That's definitely reasonable!